enhancements/olm/phone-home-support.md

```diff
@@ -0,0 +1,145 @@
+
+The purpose of this enhancement is to propose a method of communicating the health of installed operators through [Operator-Lifecycle-Manager (OLM)](https://github.com/operator-framework/operator-lifecycle-manager) to Telemeter.
+
+Today, OLM provides Telemeter with information regarding `subscription syncs` on clusters but fails to communicate the health of installed operators. By communicating which versions of operators are consistently failing, OpenShift engineers and ISV partners could proactively address issues and improve the customer experience.
+
+OLM will communicate the health of installed operators by searching for CatalogSourceVersions (CSV)s and reporting their name, phase, version, and reason fields.
+
+The initial target for this enhancement is to support reporting installed operator health per version to Telemeter in OpenShift 4.2 and 4.3.
+
+## References
+
+- https://docs.google.com/document/d/16EOP2W97EEIfkQ0FAwz3btwtMmKNDBM0qXrnDhXfzsM/edit
+- https://github.com/operator-framework/operator-lifecycle-manager/tree/master/pkg/metrics
+
+## Motivation
+
+The primary motivation for this enhancement is to enable OLM to collect information about the health per version of CSVs installed on the cluster and report these metrics to Telemeter. With this information, key insights regarding operator health could be delivered to both OpenShift Engineering as well as ISVs.
+
+### Goals
+
+- Reporting CSV health per version
+- Backport reporting CSV health per version to 4.2
+
+### Non-Goals
+
+- Reporting CSV Status Conditions per version
+- Integrating with Insights Operator in 4.3
+
+## Proposal
+
+### User Stories
+
+_**Collect CSV health per version**_
+
+As a cluster administrator, I would like OLM to expose metrics surrounding the health per version of operators deployed on the cluster so I can review them with tools such as Prometheus and Grafana.
+
+**Acceptance Criteria**
+
+OLM exposes operator health data at the metrics endpoint and users can view these metrics with tools such as Prometheus and Grafana.
+
+**Internal details**
+
+This enhancement proposes that OLM's [metrics package](https://github.com/operator-framework/operator-lifecycle-manager/tree/master/pkg/metrics) be updated to export information about the name, phase, version, and reason of CSVs deployed on cluster as a representation of operator health.
+
+Currently, information about `subscription syncs` are provided to OLM in the following format:
+
+```
+# HELP subscription_sync_total Monotonic count of subscription syncs
+# TYPE subscription_sync_total counter
+subscription_sync_total{installed="",name="akka-cluster-operator"} 10.0
+subscription_sync_total{installed="akka-cluster-operator.v0.0.1",name="akka-cluster-operator"} 1.0
+```
+
+A similar metric could be created for CSVs:
+
+```
+# HELP csv_sync_total Monotonic count of csv syncs
+# TYPE csv_sync_total counter
+csv_sync_total{name="akka", phase="Succeeded", version="0.0.1", reason="InstallSucceeded"} 1.0
```

Comment:
Does this data capture if the operand is successful? Is there some way for the operator to expose that while it may be deployed something else may be going wrong?